Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/hibernate/LoadTestDataServiceImpl.java

```diff
@@ -0,0 +1,102 @@
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package gov.hhs.fha.nhinc.admingui.hibernate;
+
+import gov.hhs.fha.nhinc.admingui.services.LoadTestDataService;
+import gov.hhs.fha.nhinc.admingui.util.HelperUtil;
+import gov.hhs.fha.nhinc.patientdb.dao.PatientDAO;
+import gov.hhs.fha.nhinc.patientdb.dao.PersonnameDAO;
+import gov.hhs.fha.nhinc.patientdb.model.Patient;
+import gov.hhs.fha.nhinc.patientdb.model.Personname;
+import java.util.List;
+import org.apache.commons.collections.CollectionUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+/**
+ *
+ * @author Tran Tang
+ */
+@Service
+public class LoadTestDataServiceImpl implements LoadTestDataService {
+    private static final Logger LOG = LoggerFactory.getLogger(LoadTestDataServiceImpl.class);
+    private PatientDAO patientDAO = PatientDAO.getPatientDAOInstance();
+    private PersonnameDAO personnameDAO = PersonnameDAO.getPersonnameDAOInstance();
+
+    @Override
+    public List<Patient> getAllPatients() {
+        LOG.info("Service-getAll-patients");
+        return patientDAO.getAll();
+    }
+
+    @Override
+    public void deletePatient(Patient patient) {
+        LOG.info("Service-delete-patient");
+    }
+
+    @Override
+    public Patient getPatientById(Long id) {
+        LOG.info("Service-getPatient-byID");
+        return patientDAO.read(id);
+    }
+
+    @Override
+    public Patient savePatient(Patient patient) {
+        LOG.info("Service-save-patient");
+
+        if (CollectionUtils.isNotEmpty(patient.getPersonnames())) {
+            LOG.debug("Personname-set-patient");
+
+            Personname personnameRecord = patient.getPersonnames().get(0);
+            personnameRecord.setPatient(patient);
+
+            if (HelperUtil.isId(patient.getPatientId())) {
+                LOG.debug("update-patient-byID");
+                patientDAO.update(patient);
+            } else {
+                LOG.debug("create-patient-new-record");
+                patientDAO.create(patient);
+            }
+
+            //patient and personname is a 1-to-1
+            LOG.debug("Personname-Patient-patientId: {} ", personnameRecord.getPatient().getPatientId());
+            if (HelperUtil.isId(personnameRecord.getPersonnameId())) {
+                LOG.debug("update-personname-byID");
+                personnameDAO.update(personnameRecord);
+            } else {
+                LOG.debug("create-personname-new-record");
+                personnameDAO.create(personnameRecord);
+            }
+        }
+        else {
+            LOG.info("save-patient-fail: personname does not exist");
+        }
+        return patient;
```

Comment:
Question: how does caller know whether the operation is successful or not? The response value is the same as the incoming request.

Reply:
well it shouldn't fail if all information is provided i can throw error if you like